src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs

```diff
@@ -539,6 +511,50 @@ private static void ExecuteAndCompleteQuery(QueryExecuteParams executeParams, Re
             }
         }
 
+        // Internal for testing purposes
+        internal string GetSqlText(ExecuteRequestParamsBase request)
+        {
+            Type requestType = request.GetType();
+
+            // If it is a document selection, we'll retrieve the text from the document
+            if (requestType == typeof(ExecuteDocumentSelectionParams))
```

Comment:
FxCop will give you warning for these two lines (typeof and casting to a type) instead you should do documentRequest =  request as ExecuteDocumentSelectionParams and then check if documentRequest  is null